src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEParameterSymbol.cs

```diff
@@ -294,16 +293,27 @@ public bool TryGetFlowAnalysisAnnotations(out FlowAnalysisAnnotations value)
                 typeWithAnnotations = NullableTypeDecoder.TransformType(typeWithAnnotations, handle, moduleSymbol, accessSymbol: accessSymbol, nullableContext: nullableContext);
                 typeWithAnnotations = TupleTypeDecoder.DecodeTupleTypesIfApplicable(typeWithAnnotations, handle, moduleSymbol);
 
-                if (_moduleSymbol.Module.HasLifetimeAnnotationAttribute(_handle, out var pair))
+                if (refKind == RefKind.Out)
+                {
+                    scope = DeclarationScope.RefScoped;
+                }
+                else if (_moduleSymbol.Module.HasScopedRefAttribute(_handle))
                 {
-                    var scopeOpt = GetScope(refKind, typeWithAnnotations.Type, pair.IsRefScoped, pair.IsValueScoped);
-                    if (scopeOpt is null)
+                    if (isByRef)
                     {
-                        isBad = true;
+                        Debug.Assert(refKind != RefKind.None);
+                        scope = DeclarationScope.RefScoped;
                     }
                     else
                     {
-                        scope = scopeOpt.GetValueOrDefault();
+                        if (typeWithAnnotations.Type.IsRefLikeType)
```

Comment:
Minor: Consider collapsing to `else if`.

Reply:
That was intentional, but I don't feel very strongly about it ;-)